toolchain/check/import_cpp.cpp

```
@@ -540,7 +540,49 @@ static auto ImportClassObjectRepr(Context& context,
   static_assert(SemIR::CustomLayoutId::FirstFieldIndex == 2);
 
   // TODO: Import vptr(s).
-  // TODO: Import bases.
+
+  // Import bases.
+  for (const auto& base : clang_def->bases()) {
+    if (base.isVirtual()) {
+      // TODO: Handle virtual bases. We don't actually know where they go in the
```

For this TODO (and maybe less so about multiple base classes), is it worth making this a `context.TODO` to flag when this path is hit? i.e., this at least seems off the beaten path, so maybe it's okay turning tests with virtual bases into failures with the TODO.

---

Yeah, silently ignoring virtual bases is a little dangerous, as they affect the size of the derived type in subtle ways. Added a `context.TODO` for that case. I'll probably revisit this, because this will block using `std::cout` from Carbon (`std::istream` has a virtual base). Maybe we can import such classes as incomplete types instead of rejecting them entirely.

For multiple inheritance, I'm inclined to not produce a diagnostic, in order to allow us to interop with more C++ code until we figure out how we want to expose such cases in Carbon. And I think the modeling we're doing in that cases isn't really wrong -- it's just that the rest of the toolchain doesn't know what to do with classes with more than one `BaseDecl` :)